src/views/pages/api-test/TestPost.js

```diff
@@ -0,0 +1,136 @@
+import { useState } from "react";
+import {
+    Box,
+    Button,
+    Card,
+    CardActions,
+    CardContent,
+    CardHeader,
+    Chip,
+    Divider,
+    MenuItem,
+    Paper,
+    Select,
+    Stack,
+    TextField,
+    Typography
+} from "@mui/material";
+
+import useApi from "hooks/useApi";
+import testApi from "api/test";
+import { styled } from "@mui/material/styles";
+import StateHandler from "ui-component/StateHandler";
+
+const Item = styled(Paper)(({ theme }) => ({
+    backgroundColor: theme.palette.mode === "dark" ? "#1A2027" : "#fff",
+    ...theme.typography.body2,
+    padding: theme.spacing(1),
+    textAlign: "center",
+    color: theme.palette.text.secondary
+}));
+
+const TestPost = () => {
+    // This API performs an operation between two numbers
+    // Parameters: n1, n2, op
+    // Supported ops: add, sub, mul, div
+    const calcApi = useApi(testApi.postTest);
+    const [calcState, setCalcState] = useState({
+        n1: "",
+        n2: "",
+        op: "add"
+    });
+
+    const handleTextInputChange = (event) => {
+        console.log("handleTextInputChange:", event.target);
+        const value = event.target.value;
+        setCalcState({
+            ...calcState,
+            [event.target.name]: value
+        });
+        console.log("calcState:", calcState);
+    };
+
+    const makeRequest = () => {
+        // Note: this artificially delays the response, to demonstrate the use of the StateHandler with Loading state
+        calcApi.requestSlow(calcState.n1, calcState.n2, calcState.op);
+    };
+
+    return (
+        <Card variant="outlined" sx={{ maxWidth: 500 }}>
+            <CardHeader title="/test_post" subheader="Test POST API Endpoint" />
+            <Divider />
+            <CardContent component={Stack} spacing={2}>
+                <Paper variant="outlined" sx={{ p: 1 }}>
+                    <Typography variant="h4" p={2}>
+                        States:
+                    </Typography>
+                    <Box textAlign="center">
+                        <Chip label={"N1: " + calcState.n1} />
+                        <Chip label={"N2: " + calcState.n2} />
+                        <Chip label={"OP: " + calcState.op} />
+                    </Box>
```

Comment:
IMPORTANT: Notice how when setCalcState is called (when you type in the input box because of the onChange), all elements that use calcState get updated automatically with the new values.